test/files/run/t9915/C_1.java

```diff
@@ -0,0 +1,18 @@
+
+public class C_1 {
+    public static final String NULLED = "X\000ABC";
```

Comment:
Well, Scala doesn't see the source-level difference; Java encodes them the same. One funny thing about your test case was that octal escapes are deprecated in Scala.